### ISSUE TITLE
Tournament bugs

### ISSUE THREAD

**Issue by KanciatyManiek** [UltraStar-Deluxe/USDX#901]
### Actual behaviour

The following bugs are currently present in Tournament mode:
- in player settings you can delete the name, but you can't enter a new one, for it to work you have to go to sing mode, set names and icons there (you can't set icons in cup mode), go to song mode to save the names players and icons,
then go back to the main screen and then you can play, because the names are displayed correctly...
![Screenshot (20241003-222937)](https://github.com/user-attachments/assets/b8aec715-3567-401f-a6c1-edb57f3aa552)
![Screenshot (20241003-223229)](https://github.com/user-attachments/assets/fc367db7-726c-450a-9904-fe6b423e884d)
- regardless of which player sings in the ladder, the icons and names of player 1 and player 2 always appear on the singing screen
![Screenshot (20241003-223305)](https://github.com/user-attachments/assets/f2731503-ba3a-4b50-9cc7-3d2d287315f3)
![Screenshot (20241003-223339)](https://github.com/user-attachments/assets/438360c4-6485-4f11-8d98-72be70bd16a8)
- In Polish, there is a typo on the song selection screen, it should be "Najlepszy" but "Najlpeszy"
![Screenshot (20241003-223442)](https://github.com/user-attachments/assets/86c6db21-29ec-4516-ae24-e7177aeeae28)

### Expected behaviour

It would be easier to play our long-time favorite game in Tournaent mode

### Details

Provide some additional information:

- USDX version: v2024.8.0
- Operating System + version: Windows 11 22631.4249

**Comment by barbeque-squared** [comment]
@KanciatyManiek for the first bug (player names can't be typed) can you check if that does work in v2023.12.0 ?
https://github.com/UltraStar-Deluxe/USDX/releases/tag/v2023.12.0 (just download the portable zip, create a `songs` folder and put some songs in it, then test if it works)

If it works on there, a fix will be easy.

The other two are probably long-standing bugs.

**Comment by KanciatyManiek** [comment]
@barbecue-squared thank you for your answer :)
I checked and in the given version 2023.12.0, entering usernames works...

The other two are probably long-lasting, but it would be nice if they could be repaired ;)

I forgot about one more error: when we press "1,2,3" to use a joker, sometimes everything is fine, and sometimes it takes two jokers at once
...The situation appeared about a year ago...

Best regards and thank you again for the great work - the game has been addictive for almost 15 years :)

**Comment by barbeque-squared** [comment]
> in the given version 2023.12.0, entering usernames works...

yeah then I know what the issue is here

> when we press "1,2,3" to use a joker, sometimes everything is fine, and sometimes it takes two jokers at once

is (if it even still happens on current versions) most likely related to what caused the username issue in the first place, thanks for reporting it!

**Comment by barbeque-squared** [comment]
* Polish translation: #907 
* Set player names: #909 
* Joker triggering twice: cannot reproduce on current version (maybe I'm doing it wrong? I don't know how these are even supposed to work)
* Player names during singing not matching the tournament names: can confirm this is a bug, but I don't really want to "fix" this by polluting the singing screen with even more party mode stuff and don't know the internals of party mode enough to say much more about it

**Comment by bohning** [comment]
Regarding the "Joker triggering twice" - since he mentions that it appeared about a year ago, could this be related to the SDL2 bug back then (IIRC, there was a general "every key press is triggered twice")?

**Comment by barbeque-squared** [comment]
Yeah that was my immediate suspicion as well. The bugged behaviour was stuff randomly triggering twice, the fixed behaviour is that it doesn't trigger twice, but from time to time some input place will surface that doesn't work at all (like this one).

Once 909 is merged I'll probably just edit the title (or create a separate issue) to be about the one remaining bug.

**Comment by KanciatyManiek** [comment]
When it comes to double Jokers, let me be more specific: I press the team number key once and the Jokers disappear twice. Sometimes setting your keyboard to the lowest key repeat value helps, but sometimes it doesn't. Interestingly, he takes double Jokers from teams 1 and 2, and takes one from team 3, as it should. This problem appeared in July 2023
, but I didn't write because I didn't have a GitHub account...

**Comment by KanciatyManiek** [comment]
> * Polish translation: [Fix typo in Polish translation #907](https://github.com/UltraStar-Deluxe/USDX/pull/907)
Great - thank you :)

> * Player names during singing not matching the tournament names: can confirm this is a bug, but I don't really want to "fix" this by polluting the singing screen with even more party mode stuff and don't know the internals of party mode enough to say much more about it
But it's not about cluttering up with new elements, because they already exist, but to incorrectly display in each duel that Player 1 and Player 2 are singing, displaying the real singing players from the ladder by downloading their correct names...
